signer/src/context/mod.rs

```diff
@@ -41,6 +45,27 @@ pub trait Context: Clone + Sync + Send {
     fn get_bitcoin_client(&self) -> impl BitcoinInteract + Clone + 'static;
     /// Get a handler to the Stacks client.
     fn get_stacks_client(&self) -> impl StacksInteract + Clone + 'static;
-    /// Get a handle to a Emily client.
+    /// Get a handle to an Emily client.
     fn get_emily_client(&self) -> impl EmilyInteract + Clone + 'static;
+    /// Create a new signal stream containing signer messages from:
+    /// 1. The signer network, as defined by the given network object
+    ///    implementing [`MessageTransfer`].
+    /// 2. The termination handled. This should only ever return one item.
+    /// 3. All messages over the signers' internal channel.
+    ///
+    /// Messages are returned as they become ready. Note that the returned
+    /// stream is not "fused", so [`StreamExt::next`] can return `None` and
+    /// later return `Some(_)`. But if [`StreamExt::next`] yields `None`
+    /// three times then the stream is "fused" and will return `None`
+    /// forever after.
+    fn new_signal_stream<M>(&self, network: &M) -> SelectAll<BroadcastStream<SignerSignal>>
```

Comment:
I think I'd call this `get_signal_stream` or something -- `new` feels like it's doing something else, where really we're just wrapping up existing streams.

Reply:
Yeah, I agree, `new` is not great. I'm sure what to do because I'm not fond of `get` either. It implies a getting a reference to a signal stream or taking something that exists, but we're resubscribing to something that exists already. I'll give it some more thought.

Reply:
Updated the name in https://github.com/stacks-network/sbtc/pull/870/commits/d7842d47921803283ea288e1734df6521407388c. I went with `as_signal_stream`... but maybe that is worse?

Reply:
I think that that's better anyway, we can nitpick it later if it bugs us :)